File: trunk/src/main/java/net/esper/eql/view/OutputCondition.java
package net.esper.eql.view;


/**
 * A condition that must be satisfied before output processing
 * is allowed to continue. Once the condition is satisfied, it
 * makes a callback to continue output processing.
 */
public interface OutputCondition {
	
	/**
	 * Update the output condition.
	 * @param newEventsCount - number of new events incoming
	 * @param oldEventsCount  - number of old events incoming
	 */ 
	public void updateOutputCondition(int newEventsCount, int oldEventsCount);
}

File: NEsper/eql/view/OutputCondition.cs
using System;
namespace net.esper.eql.view
{
	
	
	/// <summary> A condition that must be satisfied before output processing
	/// is allowed to continue. Once the condition is satisfied, it
	/// makes a callback to continue output processing.
	/// </summary>
	public interface OutputCondition
	{
		/// <summary> Update the output condition.</summary>
		/// <param name="newEventsCount">- number of new events incoming
		/// </param>
		/// <param name="oldEventsCount"> - number of old events incoming
		/// </param>
		void  updateOutputCondition(int newEventsCount, int oldEventsCount);
	}
}